src/Search/Provider/SQLProvider.php

```diff
@@ -59,11 +71,11 @@ private static function buildSelect(array $data, string $itemtable): string
     {
         // request currentuser for SQL supervision, not displayed
         $SELECT = "SELECT DISTINCT `$itemtable`.`id` AS id, '" . \Toolbox::addslashes_deep($_SESSION['glpiname'] ?? '') . "' AS currentuser,
-                        " . self::addDefaultSelect($data['itemtype']);
+                        " . \Search::addDefaultSelect($data['itemtype']);
```

Comment:
It is probably a too big step for this PR, but I think that the refactoring target should be, for the `SQLProvider` class, to be able to return array values that could be used directly in `$DB->request()` with as less as possible usage of `QueryExpression` .

Reply:
That is my goal. The next PR regarding moving from raw SQL to array/iterator would be to finalize using array values all the way through the `constructSQL` method and the methods it calls.